src/main/java/org/sonar/plugins/csharp/CSharpSensor.java

```diff
@@ -104,15 +109,47 @@ void executeInternal(SensorContext context) {
     String roslynReportPath = settings.getString(ROSLYN_REPORT_PATH_PROPERTY_KEY);
     boolean hasRoslynReportPath = roslynReportPath != null;
 
-    analyze(!hasRoslynReportPath, context);
-    importResults(context);
+    String analyzerWorkDirPath = settings.getString(ANALYZER_PROJECT_OUT_PATH_PROPERTY_KEY);
+    boolean requiresAnalyzerScannerExecution = requiresAnalyzerScannerExecution(analyzerWorkDirPath);
+
+    LOG.info("SonarAnalyzer.Scanner needs to be executed: " + requiresAnalyzerScannerExecution);
+
+    if (requiresAnalyzerScannerExecution) {
+      analyze(!hasRoslynReportPath, context);
+    }
+
+    Path workDirectory = requiresAnalyzerScannerExecution
+      ? toolOutput(context.fileSystem())
+      : analyzerOutputDir(analyzerWorkDirPath);
+
+    LOG.info("Importing analysis results from " + workDirectory.toAbsolutePath().toString());
+    importResults(context, workDirectory);
 
     if (hasRoslynReportPath) {
+      LOG.info("Importing Roslyn report");
       importRoslynReport(roslynReportPath, context);
     }
   }
 
-  private void analyze(boolean includeRules, SensorContext context) {
+  private static boolean requiresAnalyzerScannerExecution(String analyzerWorkDirPath) {
+    Path dir = analyzerOutputDir(analyzerWorkDirPath);
+    if (!Files.exists(dir)){
+      LOG.info("Analyzer working directory does not exist");
+      return true;
+    }
+
+    try {
+      DirectoryStream<Path> files = Files.newDirectoryStream(dir, p -> p.toAbsolutePath().toString().toLowerCase().endsWith(".pb"));
+      long count = StreamSupport.stream(files.spliterator(), false).count();
+      LOG.info("Analyzer working directory contains " + count + " .pb file(s)");
```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) Close this "DirectoryStream". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS2095)